lib/internal/process/promises.js

```diff
@@ -8,6 +8,11 @@ let lastPromiseId = 1;
 
 exports.setup = setupPromises;
 
+function getAsynchronousRejectionWarningObject(uid) {
+  return new Error('Promise rejection was handled ' +
+                   `asynchronously (rejection id: ${uid})`);
+}
```

Comment:
Perhaps this could reduce late promise handling performance slightly but... oh well, maybe we can make that an option if people complain. ¯\\\_(ツ)\_/¯ Good defaults are more important for this kind of thing.